INFO 2024-03-31 12:47:05,206 autoreload 9232 9632 Watching for file changes with StatReloader
WARNING 2024-03-31 12:47:11,989 log 9232 9896 Not Found: /
WARNING 2024-03-31 12:47:11,991 basehttp 9232 9896 "GET / HTTP/1.1" 404 2169
INFO 2024-03-31 12:47:17,642 views 9232 9896 Index page accessed
INFO 2024-03-31 12:47:17,646 basehttp 9232 9896 "GET /prefix/ HTTP/1.1" 200 12
INFO 2024-03-31 12:47:24,706 basehttp 9232 9896 "GET /prefix/number HTTP/1.1" 301 0
INFO 2024-03-31 12:47:24,720 basehttp 9232 5480 "GET /prefix/number/ HTTP/1.1" 200 2
INFO 2024-03-31 12:47:31,404 basehttp 9232 5480 "GET /prefix/coin HTTP/1.1" 301 0
INFO 2024-03-31 12:47:31,411 views 9232 6324 result = orel
INFO 2024-03-31 12:47:31,412 basehttp 9232 6324 "GET /prefix/coin/ HTTP/1.1" 200 4
INFO 2024-03-31 12:47:36,177 basehttp 9232 6324 "GET /prefix/cube HTTP/1.1" 301 0
INFO 2024-03-31 12:47:36,188 basehttp 9232 7320 "GET /prefix/cube/ HTTP/1.1" 200 1
INFO 2024-03-31 13:22:19,317 autoreload 1904 408 Watching for file changes with StatReloader
INFO 2024-03-31 13:25:04,714 autoreload 7792 7536 Watching for file changes with StatReloader
WARNING 2024-03-31 13:25:09,025 log 7792 8688 Not Found: /
WARNING 2024-03-31 13:25:09,031 basehttp 7792 8688 "GET / HTTP/1.1" 404 2165
INFO 2024-03-31 13:25:15,649 basehttp 7792 8688 "GET /HW HTTP/1.1" 301 0
INFO 2024-03-31 13:25:15,668 basehttp 7792 6484 "GET /HW/ HTTP/1.1" 200 353
INFO 2024-03-31 13:25:40,299 basehttp 7792 6484 "GET /HW/about_me/ HTTP/1.1" 200 549
INFO 2024-03-31 21:53:00,112 autoreload 5936 4240 Watching for file changes with StatReloader
WARNING 2024-03-31 21:53:07,571 log 5936 9816 Not Found: /
WARNING 2024-03-31 21:53:07,572 basehttp 5936 9816 "GET / HTTP/1.1" 404 2165
INFO 2024-03-31 21:53:19,184 basehttp 5936 9816 "GET /HW HTTP/1.1" 301 0
INFO 2024-03-31 21:53:19,198 basehttp 5936 5840 "GET /HW/ HTTP/1.1" 200 353
INFO 2024-03-31 21:53:37,148 basehttp 5936 5840 "GET /HW/create_client HTTP/1.1" 301 0
ERROR 2024-03-31 21:53:37,337 log 5936 7288 Internal Server Error: /HW/create_client/
Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: create_client() missing 4 required positional arguments: 'email', 'phone_number', 'address', and 'registration_date'
ERROR 2024-03-31 21:53:37,339 basehttp 5936 7288 "GET /HW/create_client/ HTTP/1.1" 500 64430
INFO 2024-03-31 21:55:07,626 basehttp 5936 7288 "GET /HW/create_client?sdfsd,sfdsdf,sdfsdf,4564 HTTP/1.1" 301 0
ERROR 2024-03-31 21:55:07,675 log 5936 1580 Internal Server Error: /HW/create_client/
Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: create_client() missing 4 required positional arguments: 'email', 'phone_number', 'address', and 'registration_date'
ERROR 2024-03-31 21:55:07,679 basehttp 5936 1580 "GET /HW/create_client/?sdfsd,sfdsdf,sdfsdf,4564 HTTP/1.1" 500 64859
INFO 2024-04-03 20:35:57,085 autoreload 8344 7108 Watching for file changes with StatReloader
INFO 2024-04-03 20:37:49,445 autoreload 3204 8208 Watching for file changes with StatReloader
WARNING 2024-04-03 20:37:52,798 log 3204 1656 Not Found: /
WARNING 2024-04-03 20:37:52,804 basehttp 3204 1656 "GET / HTTP/1.1" 404 2377
WARNING 2024-04-03 20:37:53,044 log 3204 1656 Not Found: /favicon.ico
WARNING 2024-04-03 20:37:53,046 basehttp 3204 1656 "GET /favicon.ico HTTP/1.1" 404 2428
WARNING 2024-04-03 20:38:01,020 log 3204 1656 Not Found: /les3
WARNING 2024-04-03 20:38:01,021 basehttp 3204 1656 "GET /les3 HTTP/1.1" 404 2407
WARNING 2024-04-03 20:38:21,394 log 3204 1656 Not Found: /les3/
WARNING 2024-04-03 20:38:21,394 basehttp 3204 1656 "GET /les3/ HTTP/1.1" 404 2651
INFO 2024-04-03 20:38:45,136 basehttp 3204 1656 "GET /les3/hello/ HTTP/1.1" 200 26
INFO 2024-04-03 20:38:52,163 basehttp 3204 1656 "GET /les3/hello2/ HTTP/1.1" 200 23
INFO 2024-04-03 20:39:36,018 basehttp 3204 1656 "GET /les3/hello2 HTTP/1.1" 301 0
INFO 2024-04-03 20:39:36,026 basehttp 3204 8128 "GET /les3/hello2/ HTTP/1.1" 200 23
INFO 2024-04-03 20:41:02,258 autoreload 3204 8208 D:\Education_python\Django_framework\myproject\app_lecture\urls.py changed, reloading.
INFO 2024-04-03 20:41:03,129 autoreload 7540 4044 Watching for file changes with StatReloader
INFO 2024-04-03 20:41:13,515 autoreload 7540 4044 D:\Education_python\Django_framework\myproject\app_lecture\urls.py changed, reloading.
INFO 2024-04-03 20:41:15,104 autoreload 4528 1104 Watching for file changes with StatReloader
INFO 2024-04-03 20:45:20,887 autoreload 4528 1104 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-03 20:45:22,451 autoreload 3144 7752 Watching for file changes with StatReloader
INFO 2024-04-03 20:52:29,648 autoreload 6148 5392 Watching for file changes with StatReloader
WARNING 2024-04-03 20:53:03,239 log 6148 8080 Not Found: /les3/post//python/
WARNING 2024-04-03 20:53:03,240 basehttp 6148 8080 "GET /les3/post//python/ HTTP/1.1" 404 3326
WARNING 2024-04-03 20:53:17,701 log 6148 8080 Not Found: /les3/post/2022/6/python/
WARNING 2024-04-03 20:53:17,702 basehttp 6148 8080 "GET /les3/post/2022/6/python/ HTTP/1.1" 404 3344
INFO 2024-04-03 20:53:35,370 basehttp 6148 8080 "GET /les3/posts/2022/6/python/ HTTP/1.1" 200 360
INFO 2024-04-03 21:02:22,253 autoreload 6148 5392 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-03 21:02:23,820 autoreload 3824 8096 Watching for file changes with StatReloader
INFO 2024-04-03 21:04:13,176 autoreload 3824 8096 D:\Education_python\Django_framework\myproject\app_lecture\urls.py changed, reloading.
INFO 2024-04-03 21:04:14,142 autoreload 7056 3788 Watching for file changes with StatReloader
INFO 2024-04-03 21:04:38,318 autoreload 7056 3788 D:\Education_python\Django_framework\myproject\app_lecture\urls.py changed, reloading.
INFO 2024-04-03 21:04:39,607 autoreload 5520 4140 Watching for file changes with StatReloader
INFO 2024-04-03 21:05:04,517 basehttp 5520 2472 "GET /les3/ HTTP/1.1" 200 175
INFO 2024-04-03 21:13:38,672 autoreload 5520 4140 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-03 21:13:40,085 autoreload 7820 2180 Watching for file changes with StatReloader
INFO 2024-04-03 21:14:20,423 autoreload 7820 2180 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-03 21:14:21,915 autoreload 7888 8724 Watching for file changes with StatReloader
INFO 2024-04-03 21:15:37,247 autoreload 7888 8724 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-03 21:15:38,462 autoreload 4992 7028 Watching for file changes with StatReloader
INFO 2024-04-03 21:17:17,352 autoreload 4992 7028 D:\Education_python\Django_framework\myproject\app_lecture\urls.py changed, reloading.
INFO 2024-04-03 21:17:18,271 autoreload 1960 7524 Watching for file changes with StatReloader
INFO 2024-04-03 21:23:31,739 autoreload 1960 7524 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-03 21:23:33,140 autoreload 5484 8868 Watching for file changes with StatReloader
INFO 2024-04-03 21:24:34,868 autoreload 5484 8868 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-03 21:24:36,539 autoreload 5396 7792 Watching for file changes with StatReloader
INFO 2024-04-03 21:26:04,626 autoreload 5396 7792 D:\Education_python\Django_framework\myproject\app_lecture\urls.py changed, reloading.
INFO 2024-04-03 21:26:05,847 autoreload 9036 7752 Watching for file changes with StatReloader
INFO 2024-04-03 21:26:20,491 basehttp 9036 7336 "GET /les3/for/ HTTP/1.1" 200 866
INFO 2024-04-03 21:41:28,423 autoreload 9036 7752 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-03 21:41:29,712 autoreload 4316 8872 Watching for file changes with StatReloader
INFO 2024-04-03 21:42:55,874 autoreload 4316 8872 D:\Education_python\Django_framework\myproject\app_lecture\urls.py changed, reloading.
INFO 2024-04-03 21:42:57,092 autoreload 6184 1872 Watching for file changes with StatReloader
INFO 2024-04-03 21:44:25,988 autoreload 6184 1872 D:\Education_python\Django_framework\myproject\myproject\settings.py changed, reloading.
INFO 2024-04-03 21:44:27,415 autoreload 4012 8576 Watching for file changes with StatReloader
INFO 2024-04-03 21:49:31,539 autoreload 4012 8576 D:\Education_python\Django_framework\myproject\myproject\urls.py changed, reloading.
INFO 2024-04-03 21:49:33,595 autoreload 7792 8124 Watching for file changes with StatReloader
INFO 2024-04-03 21:52:05,624 autoreload 7792 8124 D:\Education_python\Django_framework\myproject\app_lecture\models.py changed, reloading.
INFO 2024-04-03 21:52:07,134 autoreload 412 8304 Watching for file changes with StatReloader
INFO 2024-04-03 21:53:51,156 autoreload 412 8304 D:\Education_python\Django_framework\myproject\app_lecture\models.py changed, reloading.
INFO 2024-04-03 21:53:52,143 autoreload 6376 7468 Watching for file changes with StatReloader
INFO 2024-04-04 22:07:07,541 autoreload 7252 5600 Watching for file changes with StatReloader
INFO 2024-04-04 22:08:09,029 autoreload 4016 7668 Watching for file changes with StatReloader
INFO 2024-04-04 22:08:13,078 basehttp 4016 8644 "GET / HTTP/1.1" 200 313
WARNING 2024-04-04 22:08:13,617 log 4016 8644 Not Found: /favicon.ico
WARNING 2024-04-04 22:08:13,620 basehttp 4016 8644 "GET /favicon.ico HTTP/1.1" 404 2528
WARNING 2024-04-04 22:08:27,788 log 4016 8644 Not Found: /order/
WARNING 2024-04-04 22:08:27,788 basehttp 4016 8644 "GET /order/ HTTP/1.1" 404 2513
WARNING 2024-04-04 22:08:47,201 log 4016 8644 Not Found: /hw/order/
WARNING 2024-04-04 22:08:47,202 basehttp 4016 8644 "GET /hw/order/ HTTP/1.1" 404 2522
INFO 2024-04-04 22:09:09,883 basehttp 4016 8644 "GET /HW HTTP/1.1" 301 0
INFO 2024-04-04 22:09:09,897 basehttp 4016 2488 "GET /HW/ HTTP/1.1" 200 353
ERROR 2024-04-04 22:09:22,954 log 4016 2488 Internal Server Error: /HW/orders/
Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\fields\__init__.py", line 2117, in get_prep_value
    return int(value)
           ^^^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'SimpleLazyObject'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\myproject\hw_app\views.py", line 79, in orders_list
    orders = Order.objects.filter(client=request.user)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\query.py", line 1476, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\query.py", line 1494, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\query.py", line 1501, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\sql\query.py", line 1613, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\sql\query.py", line 1645, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\sql\query.py", line 1559, in build_filter
    condition = self.build_lookup(lookups, col, value)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\sql\query.py", line 1389, in build_lookup
    lookup = lookup_class(lhs, rhs)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\lookups.py", line 30, in __init__
    self.rhs = self.get_prep_lookup()
               ^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\fields\related_lookups.py", line 156, in get_prep_lookup
    self.rhs = target_field.get_prep_value(self.rhs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\fields\__init__.py", line 2119, in get_prep_value
    raise e.__class__(
TypeError: Field 'id' expected a number but got <SimpleLazyObject: <django.contrib.auth.models.AnonymousUser object at 0x000000E08777FA40>>.
ERROR 2024-04-04 22:09:22,963 basehttp 4016 2488 "GET /HW/orders/ HTTP/1.1" 500 126524
INFO 2024-04-04 22:11:14,097 autoreload 4268 1808 Watching for file changes with StatReloader
ERROR 2024-04-04 22:11:17,768 log 4268 8516 Internal Server Error: /HW/orders/
Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\myproject\hw_app\views.py", line 80, in orders_list
    return render(request, 'orders_list.html', {'orders': orders})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: orders_list.html
ERROR 2024-04-04 22:11:17,774 basehttp 4268 8516 "GET /HW/orders/ HTTP/1.1" 500 83117
INFO 2024-04-04 22:12:28,456 autoreload 4136 4168 Watching for file changes with StatReloader
INFO 2024-04-04 22:12:31,192 basehttp 4136 7988 "GET /HW/orders/ HTTP/1.1" 200 124
INFO 2024-04-04 22:14:06,609 basehttp 4136 7988 "GET /HW/orders/ HTTP/1.1" 200 136
WARNING 2024-04-04 22:14:11,960 log 4136 7988 Not Found: /HW/orders/6
WARNING 2024-04-04 22:14:11,961 basehttp 4136 7988 "GET /HW/orders/6 HTTP/1.1" 404 3492
INFO 2024-04-04 22:18:17,154 autoreload 4136 4168 D:\Education_python\Django_framework\myproject\hw_app\views.py changed, reloading.
INFO 2024-04-04 22:18:18,113 autoreload 6080 4884 Watching for file changes with StatReloader
INFO 2024-04-04 22:18:54,223 autoreload 6080 4884 D:\Education_python\Django_framework\myproject\hw_app\views.py changed, reloading.
INFO 2024-04-04 22:18:55,419 autoreload 7252 196 Watching for file changes with StatReloader
INFO 2024-04-04 22:20:34,941 autoreload 7252 196 D:\Education_python\Django_framework\myproject\hw_app\views.py changed, reloading.
INFO 2024-04-04 22:20:35,957 autoreload 7816 968 Watching for file changes with StatReloader
INFO 2024-04-04 22:25:03,197 autoreload 7816 968 D:\Education_python\Django_framework\myproject\hw_app\urls.py changed, reloading.
INFO 2024-04-04 22:25:04,169 autoreload 5016 4860 Watching for file changes with StatReloader
INFO 2024-04-04 22:26:12,181 autoreload 5016 4860 D:\Education_python\Django_framework\myproject\hw_app\urls.py changed, reloading.
INFO 2024-04-04 22:26:13,223 autoreload 6532 5408 Watching for file changes with StatReloader
WARNING 2024-04-04 22:28:04,515 log 6532 1100 Not Found: /HW/orders/6
WARNING 2024-04-04 22:28:04,516 basehttp 6532 1100 "GET /HW/orders/6 HTTP/1.1" 404 3708
INFO 2024-04-04 22:28:08,708 basehttp 6532 1100 "GET /HW/orders/ HTTP/1.1" 200 136
INFO 2024-04-04 22:28:38,599 basehttp 6532 1100 "GET /HW/ordered-products/week HTTP/1.1" 301 0
ERROR 2024-04-04 22:28:38,758 log 6532 7588 Internal Server Error: /HW/ordered-products/week/
Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\fields\__init__.py", line 2117, in get_prep_value
    return int(value)
           ^^^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'SimpleLazyObject'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\myproject\hw_app\views.py", line 92, in orders_list_time
    order_items = OrderItem.objects.filter(order__client=request.user, order__order_date__gte=start_date).order_by('order__order_date', 'product__name').distinct('product__name')
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\query.py", line 1476, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\query.py", line 1494, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\query.py", line 1501, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\sql\query.py", line 1613, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\sql\query.py", line 1645, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\sql\query.py", line 1559, in build_filter
    condition = self.build_lookup(lookups, col, value)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\sql\query.py", line 1389, in build_lookup
    lookup = lookup_class(lhs, rhs)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\lookups.py", line 30, in __init__
    self.rhs = self.get_prep_lookup()
               ^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\fields\related_lookups.py", line 156, in get_prep_lookup
    self.rhs = target_field.get_prep_value(self.rhs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\fields\__init__.py", line 2119, in get_prep_value
    raise e.__class__(
TypeError: Field 'id' expected a number but got <SimpleLazyObject: <django.contrib.auth.models.AnonymousUser object at 0x000000CE68A46000>>.
ERROR 2024-04-04 22:28:38,770 basehttp 6532 7588 "GET /HW/ordered-products/week/ HTTP/1.1" 500 128995
INFO 2024-04-04 22:30:26,397 autoreload 6532 5408 D:\Education_python\Django_framework\myproject\hw_app\views.py changed, reloading.
INFO 2024-04-04 22:30:27,620 autoreload 8808 2324 Watching for file changes with StatReloader
ERROR 2024-04-04 22:30:39,084 log 8808 9108 Internal Server Error: /HW/ordered-products/week/
Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\myproject\hw_app\views.py", line 94, in orders_list_time
    return render(request, 'hw_app/ordered_products_list.html', {'order_items': order_items})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\defaulttags.py", line 198, in render
    len_values = len(values)
                 ^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1549, in execute_sql
    sql, params = self.as_sql()
                  ^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 785, in as_sql
    distinct_result, distinct_params = self.connection.ops.distinct_sql(
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\db\backends\base\operations.py", line 205, in distinct_sql
    raise NotSupportedError(
django.db.utils.NotSupportedError: DISTINCT ON fields is not supported by this database backend
ERROR 2024-04-04 22:30:39,098 basehttp 8808 9108 "GET /HW/ordered-products/week/ HTTP/1.1" 500 133483
INFO 2024-04-04 22:34:38,137 autoreload 8808 2324 D:\Education_python\Django_framework\myproject\hw_app\views.py changed, reloading.
INFO 2024-04-04 22:34:39,229 autoreload 8180 4860 Watching for file changes with StatReloader
INFO 2024-04-04 22:35:16,299 autoreload 8180 4860 D:\Education_python\Django_framework\myproject\hw_app\views.py changed, reloading.
INFO 2024-04-04 22:35:17,374 autoreload 8060 1296 Watching for file changes with StatReloader
INFO 2024-04-04 22:35:17,757 basehttp 8060 7740 "GET /HW/ordered-products/week/ HTTP/1.1" 200 311
INFO 2024-04-04 22:35:31,035 basehttp 8060 7740 "GET /HW/orders/ HTTP/1.1" 200 136
INFO 2024-04-07 18:09:41,180 autoreload 8280 8232 Watching for file changes with StatReloader
INFO 2024-04-07 18:09:45,274 basehttp 8280 5588 "GET / HTTP/1.1" 200 313
WARNING 2024-04-07 18:09:45,701 log 8280 5588 Not Found: /favicon.ico
WARNING 2024-04-07 18:09:45,702 basehttp 8280 5588 "GET /favicon.ico HTTP/1.1" 404 2528
WARNING 2024-04-07 18:10:00,243 log 8280 5588 Not Found: /les3/users/add/
WARNING 2024-04-07 18:10:00,243 basehttp 8280 5588 "GET /les3/users/add/ HTTP/1.1" 404 4849
ERROR 2024-04-07 18:10:09,389 log 8280 5588 Internal Server Error: /les3/user/add/
Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\myproject\app_lecture\views.py", line 96, in user_form
    form.UserForm()
    ^^^^
UnboundLocalError: cannot access local variable 'form' where it is not associated with a value
ERROR 2024-04-07 18:10:09,392 basehttp 8280 5588 "GET /les3/user/add/ HTTP/1.1" 500 67194
INFO 2024-04-07 18:14:11,705 autoreload 8280 8232 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-07 18:14:12,587 autoreload 5880 8860 Watching for file changes with StatReloader
ERROR 2024-04-07 18:14:19,674 log 5880 1960 Internal Server Error: /les3/user/add/
Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\myproject\app_lecture\views.py", line 23, in user_form
    form.UserForm()
    ^^^^
UnboundLocalError: cannot access local variable 'form' where it is not associated with a value
ERROR 2024-04-07 18:14:19,678 basehttp 5880 1960 "GET /les3/user/add/ HTTP/1.1" 500 68005
INFO 2024-04-07 18:15:18,040 autoreload 5880 8860 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-07 18:15:19,040 autoreload 9108 6808 Watching for file changes with StatReloader
INFO 2024-04-07 18:15:21,972 basehttp 9108 5176 "GET /les3/user/add/ HTTP/1.1" 200 610
INFO 2024-04-07 18:20:19,268 basehttp 9108 5176 "GET /les3/user/add/ HTTP/1.1" 200 852
INFO 2024-04-07 18:23:51,312 autoreload 9108 6808 D:\Education_python\Django_framework\myproject\app_lecture\forms.py changed, reloading.
INFO 2024-04-07 18:23:52,342 autoreload 5244 4224 Watching for file changes with StatReloader
INFO 2024-04-07 18:26:18,503 autoreload 5244 4224 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-07 18:26:19,965 autoreload 5684 5588 Watching for file changes with StatReloader
INFO 2024-04-07 18:28:28,622 autoreload 5684 5588 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-07 18:28:29,963 autoreload 6136 7936 Watching for file changes with StatReloader
INFO 2024-04-07 18:30:01,900 autoreload 6136 7936 D:\Education_python\Django_framework\myproject\app_lecture\urls.py changed, reloading.
INFO 2024-04-07 18:30:02,861 autoreload 4224 5664 Watching for file changes with StatReloader
INFO 2024-04-07 18:32:21,832 basehttp 4224 8728 "GET /les3/forms/ HTTP/1.1" 200 1516
INFO 2024-04-07 18:39:48,413 autoreload 4224 5664 D:\Education_python\Django_framework\myproject\app_lecture\forms.py changed, reloading.
INFO 2024-04-07 18:39:49,567 autoreload 3424 8300 Watching for file changes with StatReloader
INFO 2024-04-07 18:41:15,485 autoreload 3424 8300 D:\Education_python\Django_framework\myproject\app_lecture\forms.py changed, reloading.
INFO 2024-04-07 18:41:16,820 autoreload 7364 5364 Watching for file changes with StatReloader
INFO 2024-04-07 18:43:40,197 autoreload 7364 5364 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-07 18:43:41,453 autoreload 7476 7672 Watching for file changes with StatReloader
INFO 2024-04-07 18:44:22,323 autoreload 7476 7672 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-07 18:44:23,568 autoreload 6868 4620 Watching for file changes with StatReloader
INFO 2024-04-07 18:45:35,290 basehttp 6868 6452 "GET /les3/forms/ HTTP/1.1" 200 2137
INFO 2024-04-07 18:46:50,611 autoreload 6868 4620 D:\Education_python\Django_framework\myproject\app_lecture\forms.py changed, reloading.
INFO 2024-04-07 18:46:51,806 autoreload 6988 4196 Watching for file changes with StatReloader
INFO 2024-04-07 18:49:44,931 autoreload 6988 4196 D:\Education_python\Django_framework\myproject\app_lecture\forms.py changed, reloading.
INFO 2024-04-07 18:49:46,087 autoreload 4128 9088 Watching for file changes with StatReloader
INFO 2024-04-07 18:50:28,533 autoreload 4128 9088 D:\Education_python\Django_framework\myproject\app_lecture\forms.py changed, reloading.
INFO 2024-04-07 18:50:29,752 autoreload 3000 6812 Watching for file changes with StatReloader
INFO 2024-04-07 18:53:48,106 autoreload 3000 6812 D:\Education_python\Django_framework\myproject\app_lecture\models.py changed, reloading.
INFO 2024-04-07 18:53:49,301 autoreload 6068 4600 Watching for file changes with StatReloader
INFO 2024-04-07 18:54:19,454 autoreload 6068 4600 D:\Education_python\Django_framework\myproject\app_lecture\models.py changed, reloading.
INFO 2024-04-07 18:54:20,407 autoreload 5980 7508 Watching for file changes with StatReloader
INFO 2024-04-07 19:35:31,955 autoreload 5192 8452 Watching for file changes with StatReloader
ERROR 2024-04-07 19:35:36,254 log 5192 2300 Internal Server Error: /les3/upload/
Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\myproject\app_lecture\views.py", line 53, in upload_image
    return render(request, 'myapp4/upload_image.html', {'form':form})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: myapp4/upload_image.html
ERROR 2024-04-07 19:35:36,259 basehttp 5192 2300 "GET /les3/upload/ HTTP/1.1" 500 84960
INFO 2024-04-07 19:36:24,819 autoreload 5192 8452 D:\Education_python\Django_framework\myproject\app_lecture\views.py changed, reloading.
INFO 2024-04-07 19:36:25,696 autoreload 3104 5808 Watching for file changes with StatReloader
INFO 2024-04-07 19:36:29,018 basehttp 3104 5052 "GET /les3/upload/ HTTP/1.1" 200 599
INFO 2024-04-07 22:16:56,041 autoreload 1916 4000 Watching for file changes with StatReloader
INFO 2024-04-07 22:17:01,109 basehttp 1916 9780 "GET / HTTP/1.1" 200 313
WARNING 2024-04-07 22:17:33,266 log 1916 9780 Not Found: /hw/edit_product/
WARNING 2024-04-07 22:17:33,267 basehttp 1916 9780 "GET /hw/edit_product/ HTTP/1.1" 404 2543
ERROR 2024-04-07 22:17:50,176 log 1916 9780 Internal Server Error: /HW/edit_product/
Traceback (most recent call last):
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\myproject\hw_app\views.py", line 121, in edit_product
    form = ProductForm()
           ^^^^^^^^^^^^^
  File "D:\Education_python\Django_framework\venv\Lib\site-packages\django\forms\models.py", line 360, in __init__
    raise ValueError("ModelForm has no model class specified.")
ValueError: ModelForm has no model class specified.
ERROR 2024-04-07 22:17:50,180 basehttp 1916 9780 "GET /HW/edit_product/ HTTP/1.1" 500 74173
INFO 2024-04-07 22:20:13,991 autoreload 1916 4000 D:\Education_python\Django_framework\myproject\hw_app\forms.py changed, reloading.
INFO 2024-04-07 22:20:14,982 autoreload 4348 9744 Watching for file changes with StatReloader
INFO 2024-04-07 22:20:21,332 basehttp 4348 8452 "GET /HW/edit_product/ HTTP/1.1" 200 1471
INFO 2024-04-07 22:21:23,964 basehttp 4348 8452 "POST /HW/edit_product/ HTTP/1.1" 200 1774
INFO 2024-04-07 22:22:48,812 basehttp 4348 8452 "GET /les3/upload/ HTTP/1.1" 200 599
INFO 2024-04-07 22:23:54,716 autoreload 4348 9744 D:\Education_python\Django_framework\myproject\hw_app\forms.py changed, reloading.
INFO 2024-04-07 22:23:55,721 autoreload 9632 9260 Watching for file changes with StatReloader
INFO 2024-04-07 22:24:02,828 autoreload 9632 9260 D:\Education_python\Django_framework\myproject\myapp\forms.py changed, reloading.
INFO 2024-04-07 22:24:03,724 autoreload 4764 8412 Watching for file changes with StatReloader
INFO 2024-04-07 22:24:48,696 autoreload 9640 9544 Watching for file changes with StatReloader
INFO 2024-04-07 22:24:52,244 basehttp 9640 1388 "GET / HTTP/1.1" 200 313
INFO 2024-04-07 22:25:01,845 basehttp 9640 1388 "GET /HW/edit_product/ HTTP/1.1" 200 1528
INFO 2024-04-07 22:25:55,715 autoreload 9640 9544 D:\Education_python\Django_framework\myproject\hw_app\forms.py changed, reloading.
INFO 2024-04-07 22:25:56,561 autoreload 1128 9572 Watching for file changes with StatReloader
INFO 2024-04-07 22:26:00,544 basehttp 1128 6452 "GET /HW/edit_product/ HTTP/1.1" 200 1545
INFO 2024-04-07 22:26:20,394 basehttp 1128 6452 "POST /HW/edit_product/ HTTP/1.1" 200 1725
INFO 2024-04-07 22:27:03,959 basehttp 1128 6452 "POST /HW/edit_product/ HTTP/1.1" 200 1725
INFO 2024-04-07 22:27:23,027 basehttp 1128 6452 "POST /HW/edit_product/ HTTP/1.1" 200 1725
